CONVERT.md

```diff
@@ -0,0 +1,301 @@
+```go
+type MongoDriverConfig struct {
+   Username                      string            // The remote database username
+   Password                      string            // The remote database password
+   ClusterEndpoint               string            // The endpoint
+   Database                      string            // The hosted database
+   Collections                   map[string]string // A mapping from a collection's 'Well Known Name' to 'Actual Name'
+   ReplicaSet                    string            // A name for the DocumentDB replica set, the empty string for non DocumentDB clusters   
+   IsStrongReadConcernEnabled    bool              // Whether a read value must be acknowledged by a majority of servers in the cluster to be returned
+   IsWriteConcernMajorityEnabled bool              // Whether a value to be written must be acknowledged by a majority of servers in the cluster before returning
+   ConnectTimeout                time.Duration     // Default timeout value to connect to a server
+   QueryTimeout   time.Duration                    // Default timeout value for a query
+
+	TLSConnectionConfig
+}
+
+type TLSConnectionConfig struct {
+   IsSSL              bool                         // Whether to use TLS in connections to the server (a major security breach not to do so)
+   VerifyCert         bool                         // When IsSSL is true, whether to validate the server's TLS certificate (another major security breach not to do so)
+   CACertChain        string                       // The Certificate Authority chain that to be used to validate the server's certificate
+   RealHostnameForSSH string                       // When using ssh to proxy to a server, as is often the case when testing locally, this can be set to the DNS name of
+                                                   // the actual server, since with ssh the server name will appear to be 'localhost'
+}
+```
+As a convenience to the applications the MongoDriverConfig struct have envconfig annotations that allow the application to use the MongoDriverConfig directly. An application is, of course, free to define its config in any way it chooses, provided it passes an instance of the above struct to the initialisation
+```go
+type MongoDriverConfig struct {
+	Username                      string            `envconfig:"MONGODB_USERNAME"    json:"-"`
+	Password                      string            `envconfig:"MONGODB_PASSWORD"    json:"-"`
+	ClusterEndpoint               string            `envconfig:"MONGODB_BIND_ADDR"   json:"-"`
+	Database                      string            `envconfig:"MONGODB_DATABASE"`
+	Collections                   map[string]string `envconfig:"MONGODB_COLLECTIONS"`
+	ReplicaSet                    string            `envconfig:"MONGODB_REPLICA_SET"`
+	IsStrongReadConcernEnabled    bool              `envconfig:"MONGODB_ENABLE_READ_CONCERN"`
```

Comment:
would suggest prod defaults here for IsStrongReadConcernEnabled & IsWriteConcernMajorityEnabled, connect timeout & query timeout